frappe/model/document.py

```diff
@@ -808,6 +811,52 @@ def _evaluate_alert(alert):
 			elif alert.event=='Method' and method == alert.method:
 				_evaluate_alert(alert)
 
+	def run_custom_server_action(self, method):
```

Comment:
move this code out of `document.py` it is very long

Reply:
moved to custom/doctype/custom_server_action/init.py